### PR TITLE
Feature/45 Right Sidebar Frontend implementation

### LINKED ISSUE #52: Composition Sidebar UI

### PR DESCRIPTION
**Pull request by emekauja**
Screenshot 2022-10-20 at 1 28 12 AM

### CODE REVIEW

File: src/components/Main.js [hunk 1]
```diff
@@ -1,57 +1,64 @@
...
-import CompositionDrawer from './views/CompositionDrawer';
+import { Sidebar } from './views/Sidebar/Sidebar';
 
 const mockModel = require('../resources/model').mockModel;
 
 const styles = () => ({
   root: {
     height: 'calc(100vh - 3.5rem)',
     width: '100%',
+    margin: 'none',
```
Comment (vidhyalongani) [inline src/components/Main.js]: @emekauja margin and padding values are not working, also i don't see the value add even id they were.

File: src/components/views/Sidebar/Sidebar.js [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+
+export const Sidebar = (props) => {
+  const { expand, setExpand, searchTerm } = props;
+
+  const renderContent = () => {
+    if (datasets?.length > 0) {
+      return (
+        <>
+          <Box paddingLeft="1.5rem" paddingY={1.5}>
+            <Typography component="body2" fontWeight={500}>
```
Comment (vidhyalongani) [inline src/components/views/Sidebar/Sidebar.js]: @emekauja body2 is a variant, not a component, hence no properties of body2 are reflecting

File: src/components/views/Sidebar/Sidebar.js [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+const sidebarOpened = true;
+
+export const Sidebar = (props) => {
+  const { expand, setExpand, searchTerm } = props;
+
+  const renderContent = () => {
+    if (datasets?.length > 0) {
+      return (
+        <>
+          <Box paddingLeft="1.5rem" paddingY={1.5}>
```
Comment (vidhyalongani) [inline src/components/views/Sidebar/Sidebar.js]: @emekauja Incorrect padding should be an px passing instead of just left. Left would 2 not 1.5, so px={2}

**Comment by vidhyalongani** [comment]
Hi @emekauja , i reviewed the UI of the sidebar, found the following issues:
Screen Shot 2022-10-26 at 2 04 06 PM

- The width is not like figma, border and border color do not match to figma and there is no box shadow in figma
Screen Shot 2022-10-26 at 2 13 49 PM

- The arrow icon is different in figma.
Screen Shot 2022-10-26 at 2 17 13 PM
- There is no space/padding on the left of the arrow icon in figma
Screen Shot 2022-10-26 at 2 30 02 PM

- Once i expand this composition, i can never collapse it. always remains expanded.

Screen Shot 2022-10-26 at 2 32 59 PM

- The child icon should be vertically aligned with the Composition text


Screen Shot 2022-10-26 at 2 37 56 PM

- The arrow icon of the child B2 should be aligned with Projection A2 icon

Screen Shot 2022-10-26 at 2 44 50 PM

- These two icons, do not have as much spacing between each other, please refer to figma again. I see extra padding left and margin-right adding this space. Overall spacing between these elements should be considered again using figma.

**Comment by vidhyalongani** [comment]
@emekauja found the following issues with font properties



Screen Shot 2022-10-26 at 2 51 45 PM

1. The overall letter spacing should be as in figma
2. The height of the row is 36 not 42. Please keep an eye on the details.



Screen Shot 2022-10-26 at 2 53 04 PM

- On hover the background color changes to white, it should remain the same.


Screen Shot 2022-10-26 at 2 56 29 PM
- The font weight of the composition heading should be 500, in figma is see 510, please use 500 instead of 600.
- remove letter spacing its not there in figma

- 
Screen Shot 2022-10-26 at 2 59 02 PM

-  use font weight same as composition heading for the labels, 500
- fix the line height
- remove letter spacing, keep it normal
Screen Shot 2022-10-26 at 3 00 51 PM

- Label is missing this 2px margin
Screen Shot 2022-10-26 at 3 02 14 PM

- Values for the line height, letter spacing, font size , font color, are not matching figma

Screen Shot 2022-10-26 at 3 05 21 PM

- From the box, flex, flex direction, align-items can be removed, not adding anything. Text overflow ellipsis should be removed, will not work on box. Ellipses should be on p tag

File: src/components/views/Sidebar/TreeView/InstanceTreeView.js [hunk 4]
```diff
@@ -0,0 +1,374 @@
...
+  const onNodeSelect = (e, nodeId) => {
+    setSelectedNodeId(nodeId);
+  };
+
+  const onNodeToggle = (e, nodeIds) => {
+    if (nodeIds.length === 0) {
+      return;
+    }
+
+    // if (nodes.length !== nodeIds.length && nodes[0] === nodeIds[0]) {
```
Comment (ddelpiano) [inline src/components/views/Sidebar/TreeView/InstanceTreeView.js]: @emekauja please remove commented code if not used

**Review by ddelpiano — CHANGES_REQUESTED** [review]
There is another problem noticed testing the sidebar.

If I click on the first composition and then try to compress this, the compression does not work unless I play with another item of the sidebar (deadlock in the state?), please fix this.

https://user-images.githubusercontent.com/37704177/199211871-a87e66f6-3dcd-4f03-8bdc-2c8d7e1a8b88.mp4

Other than this there are the changes requested by Vidhya and then this can be approved.

Thanks!

**Review by ddelpiano — APPROVED** [review]

**Pull request merged**